### PR TITLE
Tesla Gun Sprites

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
From Ausops

http://i.imgur.com/fL8rzVx.png

### CODE REVIEW

File: code/modules/projectiles/guns/energy/stun.dm [hunk 1]
```diff
@@ -6,24 +6,24 @@
...
 
+
+//Tesla Gun
 /obj/item/weapon/gun/energy/shock_revolver
-	name = "tesla revolver"
-	desc = "A high-tech revolver that fires internal, reusable shock cartridges in a revolving cylinder. The cartridges can be recharged using conventional rechargers."
-	icon_state = "stunrevolver"
-	item_state = "gun"
+	name = "tesla gun"
+	desc = "An experimental gun based on an experimental engine, it's about as likely to kill it's operater as it is the target."
```
Comment (Cheridan) [inline code/modules/projectiles/guns/energy/stun.dm]: >operater

**Comment by Cheridan** [comment]
>it's a sprite pr but it doesn't show the sprites

:do_not_litter:

File: code/modules/projectiles/guns/energy/stun.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/projectiles/guns/energy/stun.dm]: operatarerasfd

**Comment by xxalpha** [comment]
>no pictures

:put_litter_in_its_place:

File: code/modules/projectiles/guns/energy/stun.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/projectiles/guns/energy/stun.dm]: ;b whats the plan, operateives?

**Comment by PKPenguin321** [comment]
Posting sprite PRs but not the sprites is "The Kor"

**Comment by Iamgoofball** [comment]
"The Kor"
"The Goof"
"The Paprika"
"The Robustin"

**Comment by Cheridan** [comment]
http://mtgsalvation.gamepedia.com/Kor

**Comment by drynwyn** [comment]
No no, "The Robustin" is when you do everything through the web editor and have to make 6 extra commits before the travis works.

**Comment by Shapsy** [comment]
It's ausops sprites the quality is guaranteed

File: code/modules/projectiles/guns/energy/stun.dm [hunk 2]
```diff
@@ -41,6 +41,16 @@
 		tesla_zap(src, 3, 10000)
 	qdel(chain)
 
+/obj/item/weapon/gun/energy/shock_revolver/update_icon()
```
Comment (Jordie0608) [inline code/modules/projectiles/guns/energy/stun.dm]: Why snowflake this rather than change the icon and names to work with the existing code?

Comment (KorPhaeron) [inline code/modules/projectiles/guns/energy/stun.dm]: It doesn't have the little bars along the top, just an indicator light

Comment (Jordie0608) [inline code/modules/projectiles/guns/energy/stun.dm]: The existing code works with non-tiling overlays, see: laser gun and captain's laser.
The sprite does have a charge bar looking bit on it that could be easily adapted.

Comment (KorPhaeron) [inline code/modules/projectiles/guns/energy/stun.dm]: The sprites are nice now though aaaaaaa

**Comment by KorPhaeron** [comment]
@Cheridan @Jordie0608 

I went to go look at what it would require to make it work the same way as other guns and I do not want to do 40 inhand sprites. Literally 40 icons.

**Comment by Jordie0608** [comment]
That sounds like you're trying to do something very wrong there

**Comment by KorPhaeron** [comment]
@Jordie0608 

Go look a the gun inhands. You need [gun]0 through gun[5] to show different charge states, 4 directions for each of those, and then to mirror them all for the other hand (and yes I'm counting both hands, because mirroring and uploading that many pngs to the dmi is tedious work)

**Comment by Jordie0608** [comment]
Not nessecary, set an `item_state` on the gun and it'll ignore all that to just use a single sprite. You basically just then have to rename the overlays in `energy.dmi` to make this work with the existing code.

**Comment by KorPhaeron** [comment]
Stale PR I'll redo this soon maybe I hope

**Pull request closed without merging**